Google Books.js

```diff
@@ -116,17 +116,22 @@ function parseXML(text) {
 	}
 	
 	var ISBN;
-	const ISBN10Re = /(ISBN:)(\w{10})$/;
-	const ISBN13Re = /(ISBN:)(\w{13})$/;
+	const ISBN10Re = /(?:ISBN:)(\w{10})$/;
+	const ISBN13Re = /(?:ISBN:)(\w{13})$/;
+	const booksIDRe = /^(\w{12})$/;
 	var identifiers = ZU.xpath(xml, "dc:identifier", ns);
 	for (var i in identifiers) {
 		var ISBN10Match = ISBN10Re.exec(identifiers[i].textContent);
 		var ISBN13Match = ISBN13Re.exec(identifiers[i].textContent);
+		var booksIDMatch = booksIDRe.exec(identifiers[i].textContent);
 		if (ISBN10Match != null) {
-			ISBN = ISBN10Match[2];
+			ISBN = ISBN10Match[1];
 		}
 		if (ISBN13Match != null) {
-			ISBN = ISBN13Match[2];
+			ISBN = ISBN13Match[1];
+		}
+		if (booksIDMatch != null) {
+			newItem.url = 'https://books.google.com/books?id=' + booksIDMatch[1];
```

Comment:
The standard use case for the translators is to save the bibliographic metadata to make citations and list of references in publications. The citation styles are then responsible for the specific formatting in a published manuscript. It is common to show the url for webpages, blog posts and to show the doi for journal articles, conference proceeding in the list of references. However, it would be really strange if all references to books would have Google or Amazon urls. Have you ever seen a bibliography with a Google or Amazon url for a recent book?

However, I don't understand why you don't want to look at the `extra` field for the Google url. AFAIK Citoid is already parsing the `extra` field for PMID. Moreover, [we began at the WikiCite](https://meta.wikimedia.org/wiki/WikiCite_2016/Report/Group_8#Subtask.232_Improve_ID_import_into_Citoid) to put other IDs in the same way in extra fields such that Citoid can grab it from there, e.g. for the OCLC number https://github.com/zotero/translators/commit/046a7a584ca901744e74f586e3123b5eb9d7facc, https://github.com/zotero/translators/commit/ff293f90aa30334d148cbf7ece96c164b4bc98db.